TESTS/mbed_hal/sleep_manager/main.cpp

```diff
@@ -68,32 +88,22 @@ utest::v1::status_t testcase_setup(const Case *const source, const size_t index_
 {
     // Suspend the RTOS kernel scheduler to prevent interference with duration of sleep.
     osKernelSuspend();
+
```

Comment:
In the current version, only kernel is suspended. Common ticker layer is enabled and HAL ticker layer is reinitialized: `us_ticker_init();`, `lp_ticker_init();`. I think that this may cause undefined behavior. If common ticker layer is already initialized, then `lp_ticker_init()/us_ticker_init()` will disable ticker interrupt and may break the time tracing. Ticker features like `Timer` and `Timeout` assume that common ticker layer works fine are used in `test_lock_unlock_test_check()` test case.
I think that a safer method would be to completely cut off higher layers while testing HAL and rely only on the HAL ticker layer.

Reply:
Ok - aligned the PR with your proposal